### ISSUE TITLE
`rubocop --safe-auto-correct` makes syntax error

### ISSUE THREAD

**Issue by ngoto** [rubocop/rubocop#14534]
Correcting the following code by using `rubocop --safe-auto-correct`, syntax error occurred.

Code: (saved as tmp01.rb and tmp01.rb.ORIG)
```ruby
module A
  module B
    module C
      QUALITY_STRINGS =
        [ <<'_0_', <<'_1_', <<'_2_', <<'_3_', <<'_4_', <<'_5_', <<'_6_', <<'_7_', <<'_8_', <<'_9_' ].collect { |x| x.delete("\r\n").freeze }.freeze
030000,,,044400036;96662.//;7><;!!!
_0_
><<;;;;9944/!/4,,,57855!!
_1_
H????EIIIFF999;EIIBB!!
_2_
>>?EFEE?/////;:80--!!
_3_
,,,,,,,,(0,,,!.!!!!!!!!!!!!
_4_
!!!!!!!!!!!!!!!!!!
_5_
EEE@@@@EEEEE!
_6_
,,,,33336!!
_7_
HIIIII;;;;?>IECCCB/////;745=!
_8_
II;;;977FFCCC@24449?FDD!
_9_

    end
  end
end

p A::B::C::QUALITY_STRINGS
```

No .rubocop.yml existed, i.e. all configs are default.

--------

## Expected behavior

After auto correction, the ruby code can run without error.

## Actual behavior

Syntax error occurred when running corrected code.

Log:
```
$ ruby tmp01.rb
["030000,,,044400036;96662.//;7><;!!!", "><<;;;;9944/!/4,,,57855!!", "H????EIIIFF999;EIIBB!!", ">>?EFEE?/////;:80--!!", ",,,,,,,,(0,,,!.!!!!!!!!!!!!", "!!!!!!!!!!!!!!!!!!", "EEE@@@@EEEEE!", ",,,,33336!!", "HIIIII;;;;?>IECCCB/////;745=!", "II;;;977FFCCC@24449?FDD!"]
$ rubocop --debug --safe-auto-correct tmp01.rb
--safe-auto-correct is deprecated; use --autocorrect instead.
For /XXX/YYY: Default configuration from /ZZZ/3.4.6/lib/ruby/gems/3.4.0/gems/rubocop-1.80.2/config/default.yml
Use parallel by default.
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file.

Please also note that you can opt-in to new cops by default by adding this to your config:
  AllCops:
    NewCops: enable

Gemspec/AddRuntimeDependency: # new in 1.65
  Enabled: true
Gemspec/AttributeAssignment: # new in 1.77
  Enabled: true
Gemspec/DeprecatedAttributeAssignment: # new in 1.30
  Enabled: true
Gemspec/DevelopmentDependencies: # new in 1.44
  Enabled: true
Gemspec/RequireMFA: # new in 1.23
  Enabled: true
Layout/EmptyLinesAfterModuleInclusion: # new in 1.79
  Enabled: true
Layout/LineContinuationLeadingSpace: # new in 1.31
  Enabled: true
Layout/LineContinuationSpacing: # new in 1.31
  Enabled: true
Layout/LineEndStringConcatenationIndentation: # new in 1.18
  Enabled: true
Layout/SpaceBeforeBrackets: # new in 1.7
  Enabled: true
Lint/AmbiguousAssignment: # new in 1.7
  Enabled: true
Lint/AmbiguousOperatorPrecedence: # new in 1.21
  Enabled: true
Lint/AmbiguousRange: # new in 1.19
  Enabled: true
Lint/ArrayLiteralInRegexp: # new in 1.71
  Enabled: true
Lint/ConstantOverwrittenInRescue: # new in 1.31
  Enabled: true
Lint/ConstantReassignment: # new in 1.70
  Enabled: true
Lint/CopDirectiveSyntax: # new in 1.72
  Enabled: true
Lint/DeprecatedConstants: # new in 1.8
  Enabled: true
Lint/DuplicateBranch: # new in 1.3
  Enabled: true
Lint/DuplicateMagicComment: # new in 1.37
  Enabled: true
Lint/DuplicateMatchPattern: # new in 1.50
  Enabled: true
Lint/DuplicateRegexpCharacterClassElement: # new in 1.1
  Enabled: true
Lint/DuplicateSetElement: # new in 1.67
  Enabled: true
Lint/EmptyBlock: # new in 1.1
  Enabled: true
Lint/EmptyClass: # new in 1.3
  Enabled: true
Lint/EmptyInPattern: # new in 1.16
  Enabled: true
Lint/HashNewWithKeywordArgumentsAsDefault: # new in 1.69
  Enabled: true
Lint/IncompatibleIoSelectWithFiberScheduler: # new in 1.21
  Enabled: true
Lint/ItWithoutArgumentsInBlock: # new in 1.59
  Enabled: true
Lint/LambdaWithoutLiteralBlock: # new in 1.8
  Enabled: true
Lint/LiteralAssignmentInCondition: # new in 1.58
  Enabled: true
Lint/MixedCaseRange: # new in 1.53
  Enabled: true
Lint/NoReturnInBeginEndBlocks: # new in 1.2
  Enabled: true
Lint/NonAtomicFileOperation: # new in 1.31
  Enabled: true
Lint/NumberedParameterAssignment: # new in 1.9
  Enabled: true
Lint/NumericOperationWithConstantResult: # new in 1.69
  Enabled: true
Lint/OrAssignmentToConstant: # new in 1.9
  Enabled: true
Lint/RedundantDirGlobSort: # new in 1.8
  Enabled: true
Lint/RedundantRegexpQuantifiers: # new in 1.53
  Enabled: true
Lint/RedundantTypeConversion: # new in 1.72
  Enabled: true
Lint/RefinementImportMethods: # new in 1.27
  Enabled: true
Lint/RequireRangeParentheses: # new in 1.32
  Enabled: true
Lint/RequireRelativeSelfPath: # new in 1.22
  Enabled: true
Lint/SharedMutableDefault: # new in 1.70
  Enabled: true
Lint/SuppressedExceptionInNumberConversion: # new in 1.72
  Enabled: true
Lint/SymbolConversion: # new in 1.9
  Enabled: true
Lint/ToEnumArguments: # new in 1.1
  Enabled: true
Lint/TripleQuotes: # new in 1.9
  Enabled: true
Lint/UnescapedBracketInRegexp: # new in 1.68
  Enabled: true
Lint/UnexpectedBlockArity: # new in 1.5
  Enabled: true
Lint/UnmodifiedReduceAccumulator: # new in 1.1
  Enabled: true
Lint/UselessConstantScoping: # new in 1.72
  Enabled: true
Lint/UselessDefaultValueArgument: # new in 1.76
  Enabled: true
Lint/UselessDefined: # new in 1.69
  Enabled: true
Lint/UselessNumericOperation: # new in 1.66
  Enabled: true
Lint/UselessOr: # new in 1.76
  Enabled: true
Lint/UselessRescue: # new in 1.43
  Enabled: true
Lint/UselessRuby2Keywords: # new in 1.23
  Enabled: true
Metrics/CollectionLiteralLength: # new in 1.47
  Enabled: true
Naming/BlockForwarding: # new in 1.24
  Enabled: true
Naming/PredicateMethod: # new in 1.76
  Enabled: true
Security/CompoundHash: # new in 1.28
  Enabled: true
Security/IoMethods: # new in 1.22
  Enabled: true
Style/AmbiguousEndlessMethodDefinition: # new in 1.68
  Enabled: true
Style/ArgumentsForwarding: # new in 1.1
  Enabled: true
Style/ArrayIntersect: # new in 1.40
  Enabled: true
Style/BitwisePredicate: # new in 1.68
  Enabled: true
Style/CollectionCompact: # new in 1.2
  Enabled: true
Style/CollectionQuerying: # new in 1.77
  Enabled: true
Style/CombinableDefined: # new in 1.68
  Enabled: true
Style/ComparableBetween: # new in 1.74
  Enabled: true
Style/ComparableClamp: # new in 1.44
  Enabled: true
Style/ConcatArrayLiterals: # new in 1.41
  Enabled: true
Style/DataInheritance: # new in 1.49
  Enabled: true
Style/DigChain: # new in 1.69
  Enabled: true
Style/DirEmpty: # new in 1.48
  Enabled: true
Style/DocumentDynamicEvalDefinition: # new in 1.1
  Enabled: true
Style/EmptyHeredoc: # new in 1.32
  Enabled: true
Style/EmptyStringInsideInterpolation: # new in 1.76
  Enabled: true
Style/EndlessMethod: # new in 1.8
  Enabled: true
Style/EnvHome: # new in 1.29
  Enabled: true
Style/ExactRegexpMatch: # new in 1.51
  Enabled: true
Style/FetchEnvVar: # new in 1.28
  Enabled: true
Style/FileEmpty: # new in 1.48
  Enabled: true
Style/FileNull: # new in 1.69
  Enabled: true
Style/FileRead: # new in 1.24
  Enabled: true
Style/FileTouch: # new in 1.69
  Enabled: true
Style/FileWrite: # new in 1.24
  Enabled: true
Style/HashConversion: # new in 1.10
  Enabled: true
Style/HashExcept: # new in 1.7
  Enabled: true
Style/HashFetchChain: # new in 1.75
  Enabled: true
Style/HashSlice: # new in 1.71
  Enabled: true
Style/IfWithBooleanLiteralBranches: # new in 1.9
  Enabled: true
Style/InPatternThen: # new in 1.16
  Enabled: true
Style/ItAssignment: # new in 1.70
  Enabled: true
Style/ItBlockParameter: # new in 1.75
  Enabled: true
Style/KeywordArgumentsMerging: # new in 1.68
  Enabled: true
Style/MagicCommentFormat: # new in 1.35
  Enabled: true
Style/MapCompactWithConditionalBlock: # new in 1.30
  Enabled: true
Style/MapIntoArray: # new in 1.63
  Enabled: true
Style/MapToHash: # new in 1.24
  Enabled: true
Style/MapToSet: # new in 1.42
  Enabled: true
Style/MinMaxComparison: # new in 1.42
  Enabled: true
Style/MultilineInPatternThen: # new in 1.16
  Enabled: true
Style/NegatedIfElseCondition: # new in 1.2
  Enabled: true
Style/NestedFileDirname: # new in 1.26
  Enabled: true
Style/NilLambda: # new in 1.3
  Enabled: true
Style/NumberedParameters: # new in 1.22
  Enabled: true
Style/NumberedParametersLimit: # new in 1.22
  Enabled: true
Style/ObjectThen: # new in 1.28
  Enabled: true
Style/OpenStructUse: # new in 1.23
  Enabled: true
Style/OperatorMethodCall: # new in 1.37
  Enabled: true
Style/QuotedSymbols: # new in 1.16
  Enabled: true
Style/RedundantArgument: # new in 1.4
  Enabled: true
Style/RedundantArrayConstructor: # new in 1.52
  Enabled: true
Style/RedundantArrayFlatten: # new in 1.76
  Enabled: true
Style/RedundantConstantBase: # new in 1.40
  Enabled: true
Style/RedundantCurrentDirectoryInPath: # new in 1.53
  Enabled: true
Style/RedundantDoubleSplatHashBraces: # new in 1.41
  Enabled: true
Style/RedundantEach: # new in 1.38
  Enabled: true
Style/RedundantFilterChain: # new in 1.52
  Enabled: true
Style/RedundantFormat: # new in 1.72
  Enabled: true
Style/RedundantHeredocDelimiterQuotes: # new in 1.45
  Enabled: true
Style/RedundantInitialize: # new in 1.27
  Enabled: true
Style/RedundantInterpolationUnfreeze: # new in 1.66
  Enabled: true
Style/RedundantLineContinuation: # new in 1.49
  Enabled: true
Style/RedundantRegexpArgument: # new in 1.53
  Enabled: true
Style/RedundantRegexpConstructor: # new in 1.52
  Enabled: true
Style/RedundantSelfAssignmentBranch: # new in 1.19
  Enabled: true
Style/RedundantStringEscape: # new in 1.37
  Enabled: true
Style/ReturnNilInPredicateMethodDefinition: # new in 1.53
  Enabled: true
Style/SafeNavigationChainLength: # new in 1.68
  Enabled: true
Style/SelectByRegexp: # new in 1.22
  Enabled: true
Style/SendWithLiteralMethodName: # new in 1.64
  Enabled: true
Style/SingleLineDoEndBlock: # new in 1.57
  Enabled: true
Style/StringChars: # new in 1.12
  Enabled: true
Style/SuperArguments: # new in 1.64
  Enabled: true
Style/SuperWithArgsParentheses: # new in 1.58
  Enabled: true
Style/SwapValues: # new in 1.1
  Enabled: true
Style/YAMLFileRead: # new in 1.53
  Enabled: true
For more information: https://docs.rubocop.org/rubocop/versioning.html
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /XXX/YYY/tmp01.rb
F

Offenses:

tmp01.rb:4:24: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
      QUALITY_STRINGS =
                       ^
tmp01.rb:5:10: C: [Corrected] Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside array brackets.
        [ <<'_0_', <<'_1_', <<'_2_', <<'_3_', <<'_4_', <<'_5_', <<'_6_', <<'_7_', <<'_8_', <<'_9_' ].collect { |x| x.delete("\r\n").freeze }.freeze
         ^
tmp01.rb:5:121: C: [Corrected] Layout/LineLength: Line is too long. [147/120]
        [ <<'_0_', <<'_1_', <<'_2_', <<'_3_', <<'_4_', <<'_5_', <<'_6_', <<'_7_', <<'_8_', <<'_9_' ].collect { |x| x.delete("\r\n").freeze }.freeze
                                                                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
tmp01.rb:6:1: C: [Corrected] Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<.
030000,,,044400036;96662.//;7><;!!! ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tmp01.rb:8:1: C: [Corrected] Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<.
><<;;;;9944/!/4,,,57855!! ...
^^^^^^^^^^^^^^^^^^^^^^^^^
tmp01.rb:10:1: C: [Corrected] Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<.
H????EIIIFF999;EIIBB!! ...
^^^^^^^^^^^^^^^^^^^^^^
tmp01.rb:12:1: C: [Corrected] Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<.
>>?EFEE?/////;:80--!! ...
^^^^^^^^^^^^^^^^^^^^^
tmp01.rb:14:1: C: [Corrected] Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<.
,,,,,,,,(0,,,!.!!!!!!!!!!!! ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^
tmp01.rb:16:1: C: [Corrected] Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<.
!!!!!!!!!!!!!!!!!! ...
^^^^^^^^^^^^^^^^^^
tmp01.rb:18:1: C: [Corrected] Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<.
EEE@@@@EEEEE! ...
^^^^^^^^^^^^^
tmp01.rb:20:1: C: [Corrected] Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<.
,,,,33336!! ...
^^^^^^^^^^^
tmp01.rb:22:1: C: [Corrected] Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<.
HIIIII;;;;?>IECCCB/////;745=! ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tmp01.rb:24:1: C: [Corrected] Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<.
II;;;977FFCCC@24449?FDD! ...
^^^^^^^^^^^^^^^^^^^^^^^^
tmp01.rb:26:1: C: [Corrected] Layout/EmptyLinesAroundModuleBody: Extra empty line detected at module body end.
tmp01.rb:27:5: F: Lint/Syntax: unexpected token kEND
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
    end
    ^^^
tmp01.rb:32:1: F: Lint/Syntax: unexpected token $end
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)

1 file inspected, 16 offenses detected, 14 offenses corrected
Finished in 0.30952743702800944 seconds
$ ruby tmp01.rb
tmp01.rb: --> tmp01.rb

Unmatched `{', missing `}' ?

   1  module A
   2    module B
>  3      module C
>  4        QUALITY_STRINGS =
>  5          [<<~'_0_', <<~'_1_', <<~'_2_', <<~'_3_', <<~'_4_', <<~'_5_', <<~'_6_', <<~'_7_', <<~'_8_', <<~'_9_'].collect { |x|
>  6   x.delete("\r\n").freeze }.freeze
>  7  030000,,,044400036;96662.//;7><;!!!
>  8  _0_
>  9  ><<;;;;9944/!/4,,,57855!!
> 10  _1_
> 11  H????EIIIFF999;EIIBB!!
> 12  _2_
> 13  >>?EFEE?/////;:80--!!
> 14  _3_
> 15  ,,,,,,,,(0,,,!.!!!!!!!!!!!!
> 16  _4_
> 17  !!!!!!!!!!!!!!!!!!
> 18  _5_
> 19  EEE@@@@EEEEE!
> 20  _6_
> 21  ,,,,33336!!
> 22  _7_
> 23  HIIIII;;;;?>IECCCB/////;745=!
> 24  _8_
> 25  II;;;977FFCCC@24449?FDD!
> 26  _9_
> 27      end
  28    end
  29  end

tmp01.rb:27: syntax errors found (SyntaxError)
   3 |     module C
   4 |       QUALITY_STRINGS =
>  5 | ...
     |     ^ expected a block beginning with `{` to end with `}`
   6 |  x.delete("\r\n").freeze }.freeze
   7 | 030000,,,044400036;96662.//;7><;!!!
  ~~~~~~
  25 | II;;;977FFCCC@24449?FDD!
  26 | _9_
> 27 |     end
     |     ^~~ unexpected 'end', assuming it is closing the parent module definition
  28 |   end
  29 | end

$ diff -U0 tmp01.rb.ORIG tmp01.rb
--- tmp01.rb.ORIG       2025-09-18 17:06:31.582891000 +0900
+++ tmp01.rb    2025-09-18 17:07:28.055305000 +0900
@@ -4,2 +4,3 @@
-      QUALITY_STRINGS =
-        [ <<'_0_', <<'_1_', <<'_2_', <<'_3_', <<'_4_', <<'_5_', <<'_6_', <<'_7_', <<'_8_', <<'_9_' ].collect { |x| x.delete("\r\n").freeze }.freeze
+      QUALITY_STRINGS =
+        [<<~'_0_', <<~'_1_', <<~'_2_', <<~'_3_', <<~'_4_', <<~'_5_', <<~'_6_', <<~'_7_', <<~'_8_', <<~'_9_'].collect { |x|
+ x.delete("\r\n").freeze }.freeze
@@ -26 +26,0 @@
-
```

## Steps to reproduce the problem

Please see tha above log.

## RuboCop version

```
$ rubocop -V
1.80.2 (using Parser 3.3.9.0, rubocop-ast 1.46.0, analyzing as Ruby 2.7, running on ruby 3.4.6) [x86_64-linux]
$ ruby -v
ruby 3.4.6 (2025-09-16 revision dbd83256b1) +PRISM [x86_64-linux]
```
